### PR TITLE
Issue#33

### PR DESCRIPTION
**Pull request by csamiselo**
Changes made: Form Validation and Male Patient registration issue

### CODE REVIEW

File: common/src/main/java/zm/gov/moh/common/submodule/form/view/FormFragment.java [hunk 1]
```diff
@@ -88,6 +89,25 @@ public View onCreateView (LayoutInflater inflater, ViewGroup container,
...
+
+                if(logic.getAction().getType().equals(Action.ACTION_TYPE_CRITERIA)){
+
+                    for(String tag : logic.getAction().getMetadata().getTags())
+                        if(bundle.containsKey(tag)) {
+
+                            String value = bundle.getString(tag);
+                            if(!value.equals(logic.getCondition().getValue())){
+                                context.onBackPressed();
+                                Toast.makeText(context,"Male Patients cannnot enroll for this service",Toast.LENGTH_SHORT).show();
```
Comment (hallix) [inline common/src/main/java/zm/gov/moh/common/submodule/form/view/FormFragment.java]: @csamiselo 
1. Correct typo `cannnot` 
3. Lower case letter `p` on `Patients`
3. Move String literal `"Male Patients cannnot enroll for this service"` to XML resources incase a requirement for multi language support arises.

File: cervicalcancer/src/main/assets/forms/treatment_cryo_reproductive_health.json [hunk 2]
```diff
@@ -56,11 +56,11 @@
...
         },
         {
-          "tag": "last menstrual period",
+          "tag": "last_menstrual_period",
           "conceptId": 165202,
-          "widgetType": "DatePickerButton",
+          "widgetType": "DatePicker",
           "weight": 1,
-          "text": "Date of Last Menstrual Period ",
+          "hint": "Date of Last Menstrual Period ",
```
Comment (hallix) [inline cervicalcancer/src/main/assets/forms/treatment_cryo_reproductive_health.json]: @csamiselo 
Add text describing purpose of the date picker in the `label` attribute support for that has been added on the commit:
https://github.com/BlueCodeSystems/smartcerv/pull/58/commits/18acc79e6cab244f75cbde1eb4b2ca71584cecc0 
Includes an example of how to declare the widget data.

File: common/src/main/java/zm/gov/moh/common/submodule/form/view/FormFragment.java (hunk 1 above)
Comment (csamiselo) [reply common/src/main/java/zm/gov/moh/common/submodule/form/view/FormFragment.java]: Alright will make amendments

File: cervicalcancer/src/main/assets/forms/treatment_cryo_reproductive_health.json (hunk 2 above)
Comment (csamiselo) [reply cervicalcancer/src/main/assets/forms/treatment_cryo_reproductive_health.json]: noted

**Pull request merged**